### DISCUSSION TITLE
Trigger an error in the router

### DISCUSSION THREAD

**Discussion by WebVPF** [wintercms/winter#815] [Q&A] [answered]
How to correctly cause an error in the router?

The documentation has this: https://wintercms.com/docs/services/router#throwing-404-errors

My code:

```php
<?php

Route::get('test.txt', function ()
{
    if ( Settings::get('active') ) {
        // ...
    }
    else {
        App::abort(404);
    }
});
```

I get an exception (application error).

![2023-01-07_025945](https://user-images.githubusercontent.com/61043464/211123891-0dea977a-d68e-4643-a85f-123ac336d62e.jpg)

**Comment by alisaracoglu** [comment]
This error comes because of `App::abort(404);` row. So `Settings::get('active')` return not true.

**Reply by WebVPF** [reply]
Everything is correct. If the setting is disabled, you need to raise an error.
In my case, the settings are disabled. I need to display a 404 error.

**Reply by alisaracoglu** [reply]
First of all, you return an http 404 with `App::abort(404)`. your question may have different answers depending on your general code situation. For example; If you do not have a 404 page and forwarding is not suitable for you, you can add a view to present the 404 message here. or you can redirect to an existing 404 page. Or you can return an appropriate response if your system is just an api service. which one is more suitable for you

**Comment by LukeTowers** [comment, marked as answer]
If you would like to return a rendered 404 page as seen in the backend then you can do the following:

```php
return Response::make(View::make('backend::404'), 404);
```

If you would like to return the 404 page as configured in the frontend then you can do the following:
```php
$controller = new Cms\Classes\Controller();
$controller->setStatusCode(404);
return $controller->run('404');
```

**Reply by WebVPF** [reply]
Thanks!
